src/service.js

```diff
@@ -83,20 +88,27 @@ LS2Request.prototype.send = function() {
 		}
 		if(!self.subscribe) {
 			self.cancel();
+		} else if(self.ts && refs[self.ts]) {
```

Comment:
Removing the ref doesn't kill or end or delete the LS2Request object, however the GC may act on it.  To 'cancel' a subscription, the LS2Request function cancel() needs to be run, so it would make sense for an app to hold the request object reference within the app somewhere.

The easy solution would be to always hold on to the subscription LS2Request references to prevent their gc, however that could easily lead to memory leaks, given finite subscription services. For example, if a download service reported a file download % every second and finished at 100%. we wouldn't want to hold on to that reference beyond its lifespan.

We could enforce that within Enyo 3.x that the cancel() function is called during some unsubscribe process.   I'm not certain if there is some significant way to determine a final service subscription callback being executed (such as a specific returnValue or identifier). We may need to ask someone on the WebAppMgr or system services teams.